crates/uv/tests/pip_install.rs

```diff
@@ -3517,60 +3517,65 @@ fn already_installed_local_version_of_remote_package() {
 #[test]
 #[cfg(unix)]
 fn already_installed_multiple_versions() -> Result<()> {
-    use crate::common::copy_dir_all;
+    fn prepare(context: &TestContext) -> Result<()> {
+        use crate::common::copy_dir_all;
+
+        // Install into the base environment
+        context.install().arg("anyio==3.7.0").assert().success();
+
+        // Install another version into another environment
+        let context_duplicate = TestContext::new("3.12");
+        context_duplicate
+            .install()
+            .arg("anyio==4.0.0")
+            .assert()
+            .success();
+
+        // Copy the second version into the first environment
+        copy_dir_all(
+            context_duplicate
+                .site_packages()
+                .join("anyio-4.0.0.dist-info"),
+            context.site_packages().join("anyio-4.0.0.dist-info"),
+        )?;
+
+        Ok(())
+    }
 
-    // Install a version
-    let context1 = TestContext::new("3.12");
-    uv_snapshot!(context1.filters(), context1.install()
-        .arg("anyio==3.7.0"), @r###"
-    success: true
-    exit_code: 0
-    ----- stdout -----
+    let context = TestContext::new("3.12");
 
-    ----- stderr -----
-    Resolved 3 packages in [TIME]
-    Downloaded 3 packages in [TIME]
-    Installed 3 packages in [TIME]
-     + anyio==3.7.0
-     + idna==3.6
-     + sniffio==1.3.1
-    "###
-    );
+    prepare(&context)?;
 
-    // Install another version
-    let context2 = TestContext::new("3.12");
-    uv_snapshot!(context2.filters(), context2.install()
-        .arg("anyio==4.0.0"), @r###"
+    // Request the second anyio version again
+    // Should remove both previous versions and reinstall the second one
+    uv_snapshot!(context.filters(), context.install().arg("anyio==4.0.0"), @r###"
     success: true
     exit_code: 0
     ----- stdout -----
 
     ----- stderr -----
     Resolved 3 packages in [TIME]
-    Downloaded 3 packages in [TIME]
-    Installed 3 packages in [TIME]
+    Downloaded 1 package in [TIME]
+    Installed 1 package in [TIME]
+     - anyio==3.7.0
+     - anyio==4.0.0
      + anyio==4.0.0
-     + idna==3.6
-     + sniffio==1.3.1
     "###
     );
 
-    // Copy the second version into the first environment
-    copy_dir_all(
-        context2.site_packages().join("anyio-4.0.0.dist-info"),
-        context1.site_packages().join("anyio-4.0.0.dist-info"),
-    )?;
+    // Reset the test context
+    prepare(&context)?;
 
-    // Request the second anyio version again
-    // Should remove both previous versions and reinstall the second one
-    uv_snapshot!(context1.filters(), context1.install().arg("anyio==4.0.0"), @r###"
+    // Request the anyio without a version specifier
+    // This is loosely a regression test for the ordering of the installation preferences
+    // from existing site-packages
```

Comment:
Before sorting the site-packages, this preferred `anyio==3.7.0` on macOS and `anyio==4.0.0` on Ubuntu